File: solutions/cpp/sublist/1/sublist.cpp
#include "sublist.h"
#include <vector>
#include <algorithm>

namespace sublist {

bool is_sublist(const std::vector<int>& shorter, const std::vector<int>& longer){
        auto it_last = end(longer) -  size(shorter)+1;
        for (auto it=begin(longer); it!=it_last; ++it){
            if(std::equal(begin(shorter), end(shorter),it))
                return true;
        }
    return false;
}    
   
sublist::List_comparison sublist(const std::vector<int> &one, const std::vector<int> &two){
    List_comparison result;
    if (size(one)<size(two) && is_sublist(one,two)) 
        result = List_comparison::sublist;
    else if (size(one)>size(two) && is_sublist(two,one)) 
        result = List_comparison::superlist;
    else if (one==two) 
        result = List_comparison::equal;
    else
        result = List_comparison::unequal;
    return result;    
}
    
}  // namespace sublist

File: solutions/dart/sublist/1/lib/sublist.dart
import 'package:collection/collection.dart';

enum Classification {unequal, sublist, superlist, equal}

class Sublist {

  bool isSublist(List<int> short, List<int> long){
    for(var i=0; i<(long.length-short.length)+1; i++){
      if(ListEquality().equals(
        long.getRange(i, i+short.length).toList()
          , short))
        return true;
    } 
    return false;
  }
  
  Classification sublist(List<int> one, List<int> two){
    if(one.length<two.length && isSublist(one,two))
      return Classification.sublist;
    else if(one.length>two.length && isSublist(two,one))
      return Classification.superlist;
    else if(ListEquality().equals(one, two))
      return Classification.equal;
    else
      return Classification.unequal;
  }
}
